app/routes/author_routes.py

```diff
@@ -1,52 +1,27 @@
-from flask import Blueprint, request, make_response, abort
+from flask import Blueprint, request
 from app.models.author import Author
 from app.models.book import Book
-from .route_utilities import validate_model
-from ..db import db
+from .route_utilities import create_model, get_models_with_filters, validate_model
 
 bp = Blueprint("authors_bp", __name__, url_prefix="/authors")
 
 @bp.post("")
 def create_author():
     request_body = request.get_json()
-
-    try:
-        new_author = Author.from_dict(request_body)
-        
-    except KeyError as error:
-        response = {"message": f"Invalid request: missing {error.args[0]}"}
-        abort(make_response(response, 400))
-    
-    db.session.add(new_author)
-    db.session.commit()
-
-    return make_response(new_author.to_dict(), 201)
+    return create_model(Author, request_body)
 
 @bp.get("")
 def get_all_authors():
-    query = db.select(Author)
-
-    name_param = request.args.get("name")
-    if name_param:
-        query = query.where(Author.name.ilike(f"%{name_param}%"))
-
-    authors = db.session.scalars(query.order_by(Author.id))
-    authors_response = [author.to_dict() for author in authors]
-
-    return authors_response
+    filters = request.args
```

Comment:
I think it's a little aspirational to call this `filters`. After all, it's whatever query parameters were provided, which might contain stuff that's not filter-compatible. We'll probably need to be a little careful about how we discuss `get_models_with_filters` since it's currently a little brittle with respect to column names.

Reply:
That's fair, I can switch it to something like `query_params` to better reflect the contents.

Reply:
Are you thinking that the definition for `get_models_with_filters` and the param there should also be updated to something like `queries` rather than `filters`?

Reply:
I'm good passing `request.args` directly since we don't have any other function calls or nested access happening on that return line.